docs/design/proposals/provisioned-service/provisioned-service.md

```diff
@@ -0,0 +1,108 @@
+Kubernetes makes application connectivity to ACK-backed services easy
+and seamless.
+
+## Existing Solutions
+
+AWS Controllers for Kubernetes have support for a [FieldExport
+API](https://github.com/aws-controllers-k8s/community/blob/main/docs/design/proposals/native-binding/native-binding.md).
+This can be used to create a Secret that can be used for [Direct Secret
+Reference as per the
+specification](https://github.com/servicebinding/spec#direct-secret-reference).
+This solution needs more effort from the application developers.
+
+The FieldExports resources required for application developers are the
+same, but every developer needs to repeat this same configuration for
+every application they deploy. The proposed solution avoids this
+repetition through Provisioned Services.
+
+## Proposed Solution
+
+The proposed solution is to make various ACK resources become
+Provisioned Services. For example, make the DBInstance resource provided
+by the RDS controller a Provisioned Service. Each controller would use
+their main custom resource as a Provisioned Service resource. For each
+Provisioned Service, identify the fields that need to be exported to
+create the Secret resource.
+
+The application developer experience is explained in this demo video:
+[https://www.youtube.com/watch?v=AXXWv7N12JM](https://www.youtube.com/watch?v=AXXWv7N12JM)
+
+(This is created as part of a [Proof of
+Concept](https://github.com/aws-controllers-k8s/community/issues/1289))
+
+The structure change for DBInstance:
+
+```
+// ServiceBindingSecretReference defines a mirror of corev1.LocalObjectReference
+type ServiceBindingSecretReference struct {
+	// Name of the referent secret.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+	Name string `json:"name"`
+}
+
+// DBInstanceStatus defines the observed state of DBInstance
+type DBInstanceStatus struct {
+	// Binding exposes the Secret for Service Binding to conform the Provisioned Service
+	// as per the Service Binding Specification for Kubernetes.
+	// Ref. https://github.com/servicebinding/spec#provisioned-service
+	Binding *ServiceBindingSecretReference `json:"binding,omitempty"`
+```
+
+### Benefits
+
+When there are many microservices connecting to the same service, the
+label selector feature helps to easily bind all of them with minimal
+configuration. Service Binding Specification standard provides a uniform
+experience for all the applications connectivity with services.
+
+### Implementation
+
+Since the API changes required for various custom resources are
+standard, and they can be generated for all controllers. However, the
+changes for each of these controllers can be implemented gradually.
+
+### RBAC
+
+Since a Secret resource needs to be created, updated, and deleted, the
+ClusterRole requires "*create",* "*update", and "delete"* permissions.
+Currently, this permission is not set in the existing controllers.
```

Comment:
The [current `ClusterRole`](https://github.com/aws-controllers-k8s/s3-controller/blob/01c6f0aa4a8a1b33c8c552242affcdbc8f4df651/helm/templates/cluster-role-controller.yaml#L32-L40) allows getting, patching, listing and watching Secret resources. Why would we need to add Create and Delete permissions?

Reply:
The Service Binding Specification expects the Provisioned Service (e.g., `DBInstance`) to point to a Secret resource with all values required for connectivity from the application. In the case of `DBInstance`, the corresponding controller needs to ensure the presence of the Secret resource. To create a Secret, `Create` permission is required. Similarly when the `DBInstance` is getting deleted, cleaning up the corresponding Secret resource would be good. Hence the `Delete` permission.

Reply:
@baijum how does the controller know what the Secret's name will be?

Reply:
The `DBInstance` controller generates the Secret. For example, the name could be the same as the `DBInstance` name.